lib/setupHooks/removeReferencesToRustToolchain.sh

```diff
@@ -0,0 +1,24 @@
+# Strip any references to the rust toolchain (which can slip in through stdlib panic info) to avoid 
+# including the toolchain in the runtime closure
+
+removeReferencesToRustToolchain() {
+    local installLocation="${1:-${out:?not defined}}"
+    # NOTE(adam.dayan): would it be better to explicitly point to $(rustc --print sysroot)/lib/rustlib/src so we only scrub references to
+    # the src? are there any reasons one might want a reference to the rest of the toolchain?
```

Comment:
I'd say we don't need to worry about this hook being perfectly sophisticated for every possible use case. We should focus on removing references to the Rust toolchain's store path, and if that's not quite right for a particular project, they can disable the hook and tailor their own reference handling!

```suggestion

```